statsmodels/regression/_tools.py

```diff
@@ -22,24 +19,34 @@ class _MinimalWLS(object):
         1d array of weights.  If you supply 1/W then the variables are pre-
         multiplied by 1/sqrt(W).  If no weights are supplied the default value
         is 1 and WLS reults are the same as OLS.
+    check_finite : bool, optional
+        Flag indicating whether to check for inf/nan in endog and weights.
+        If True and any are found, ValueError is raised.
 
     Notes
     -----
-    Need resid, scale, fittedvalues, model.weights!
+    Need resid, scale, fittedvalues, model.weights
         history['scale'].append(tmp_results.scale)
         if conv == 'dev':
             history['deviance'].append(self.deviance(tmp_results))
         elif conv == 'sresid':
             history['sresid'].append(tmp_results.resid/tmp_results.scale)
         elif conv == 'weights':
             history['weights'].append(tmp_results.model.weights)
-    Does not perform and checks on the input data
+    Does not perform any checks on the input data
     """
 
-    def __init__(self, endog, exog, weights=1.0):
+    def __init__(self, endog, exog, weights=1.0, check_finite=False):
         self.endog = endog
         self.exog = exog
         self.weights = weights
+        if check_finite:
+            msg = 'NaN/inf detected in {0}, estimation infeasible.'
+            if not np.all(np.isfinite(endog)):
```

Comment:
the fastest way to check this is to use `np.isfinite(endog.ptp())` according to a stackoverflow (?) question
(the only disadvantage is that it causes a numpy warning if there is a nan or inf, e.g. in the unit tests)
`np.all(np.isfinite(` needs to loop through the array twice and needs to create an intermediate elementwise array

Reply:
```
%timeit np.all(np.isfinite(endog))
748 µs ± 33.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit np.isfinite(endog.ptp())
1.02 ms ± 47.8 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Reply:
```
%timeit np.isfinite(endog)
641 µs ± 24.1 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit endog.ptp()
982 µs ± 7.85 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit endog.sum()
924 µs ± 8.6 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Reply:
`ptp` is slow because it loops across the entire array twice, once for the max and once for the min.

https://github.com/numpy/numpy/blob/9d0225b800df3c2b0bffee9960df87b15527509c/numpy/core/src/multiarray/calculation.c#L321

Reply:
that's what I get in my older setup

```
x = np.random.randn(100000, 5)
%timeit np.all(np.isfinite(x))
%timeit np.isfinite(x.ptp())
1000 loops, best of 3: 962 µs per loop
1000 loops, best of 3: 227 µs per loop
```

a quick stackoverflow search finds https://stackoverflow.com/questions/6736590/fast-check-for-nan-in-numpy
but I don't remember from where I got the `ptp` part

Reply:
On your example

```
%timeit np.all(np.isfinite(x))
283 µs ± 5 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit np.isfinite(x.ptp())
365 µs ± 5.35 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```